docs/src/rule_author/writing_good_rules.md

```diff
@@ -71,6 +71,13 @@ Examples being:
 - There is only one derivative being returned, so from the fact that the user called
   `frule`/`rrule` they clearly will want to use that one.
 
+!! Warning
+   If your rule returns the tangent it receives without change (such as `rrule(::typeof(+), xs::AbstractArray)`
+   or some view / wrapper of this (such as `rrule(::typeof(adjoint), x::AbstractMatrix)`
+   then it is unsafe to wrap it in a `@thunk`. This is because gradient accumulation assumes that
+   the result of `unthunk`ing may be mutated for efficiency, but if the same array appears elsewhere,
+   this will give wrong answers.
```

Comment:
Not sure this is the ideal place to add this warning. But it does need to be noted somewhere.

Reply:
yes. 
Also probably worth saying "unsafe (and unnesciary as no computation is being done)"

I need to think about this hard still.
It relates to the linear nature of pullback that should give us some promises that some things are safe.
But nailing those promises down is hard because julia allows aliasing.
I was reminded